src/VkBootstrap.h

```diff
@@ -419,16 +445,31 @@ class PhysicalDeviceSelector {
 	// Require a physical device that supports a (major, minor) version of vulkan.
 	PhysicalDeviceSelector& set_minimum_version(uint32_t major, uint32_t minor);
 
-	// Require a physical device which supports the features in VkPhysicalDeviceFeatures.
-	PhysicalDeviceSelector& set_required_features(VkPhysicalDeviceFeatures const& features);
-
+	// Require a physical device which supports a specific set of general/extension features.
+#if defined(VK_API_VERSION_1_1)
+    template <typename T>
+    PhysicalDeviceSelector& add_required_extension_features(T const& features) {
+		assert(features.sType != 0 &&
+		       "Features struct sType must be filled with the struct's corresponding VkStructureType enum");
+		assert(features.sType != VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2 &&
+		       "Do not pass VkPhysicalDeviceFeatures2 as a required extension feature structure. An instance of this is managed internally for selection criteria and device creation.");
+		detail::GenericFeaturesPNextNode node;
+		node.set(features);
+        criteria.extended_features_chain.push_back(node);
+        return *this;
+    }
+#endif
+    PhysicalDeviceSelector& set_required_features(VkPhysicalDeviceFeatures const& features) {
```

Comment:
Accidentally dropped the comment before this function.
`// Require a physical device which supports the features in VkPhysicalDeviceFeatures.`
Also would be nice to put the implementation back in the cpp file so the header is mostly only function declarations.